src/sn_utils.c

```diff
@@ -346,30 +352,46 @@ static int process_mgmt(n2n_sn_t *sss,
                         (long unsigned int)(now - sss->stats.last_reg_super));
 
     ressize += snprintf(resbuf+ressize, N2N_SN_PKTBUF_SIZE-ressize,
-                        "cur_cmnts");
+                        "cur_cmnts %u\n", HASH_COUNT(sss->communities));
     HASH_ITER(hh, sss->communities, community, tmp) {
-      ressize += snprintf(resbuf+ressize, N2N_SN_PKTBUF_SIZE-ressize,
-                          " [%s]",
-                          community->community);
+      ressize += snprintf(resbuf + ressize, N2N_SN_PKTBUF_SIZE - ressize,
+                          "community: %s\n", community->community);
+      sendto_mgmt(sss, sender_sock, resbuf, ressize);
+      ressize = 0;
+
+      num = 0;
       HASH_ITER(hh, community->edges, peer, tmpPeer) {
-        ressize += snprintf(resbuf+ressize, N2N_SN_PKTBUF_SIZE-ressize,
-                            " {%s}",
-                            macaddr_str(mac_buf, peer->mac_addr));
+        ressize += snprintf(resbuf + ressize, N2N_SN_PKTBUF_SIZE - ressize,
+                            "\t[id: %u][MAC: %s][edge: %s][last seen: %lu sec ago]\n",
+                            ++num, macaddr_str(mac_buf, peer->mac_addr),
+                            sock_to_cstr(sockbuf, &(peer->sock)), now-peer->last_seen);
+
+        sendto_mgmt(sss, sender_sock, resbuf, ressize);
+        ressize = 0;
       }
     }
+
     ressize += snprintf(resbuf+ressize, N2N_SN_PKTBUF_SIZE-ressize,
                         "\n");
+    sendto_mgmt(sss, sender_sock, resbuf, ressize);
 
-    r = sendto(sss->mgmt_sock, resbuf, ressize, 0 /*flags*/,
-               (struct sockaddr *)sender_sock, sizeof(struct sockaddr_in));
+    return 0;
+}
 
-    if (r <= 0)
-    {
-        ++(sss->stats.errors);
-        traceEvent(TRACE_ERROR, "process_mgmt : sendto failed. %s", strerror(errno));
-    }
+static int sendto_mgmt(n2n_sn_t *sss,
+                       const struct sockaddr_in *sender_sock,
+                       const uint8_t *mgmt_buf,
+                       size_t mgmt_size)
+{
+  ssize_t r = sendto(sss->mgmt_sock, mgmt_buf, mgmt_size, 0 /*flags*/,
+                     (struct sockaddr *)sender_sock, sizeof (struct sockaddr_in));
 
-    return 0;
+  if (r <= 0) {
+    ++(sss->stats.errors);
+    traceEvent (TRACE_ERROR, "process_mgmt : sendto failed. %s", strerror (errno));
```

Comment:
An absolutely minor comment (of which I am about to be sorry to even make it :smile:) : With the new `sendto_mgmt()` in place, we could replace trace output `process_mgmt…` with `sendto_mgmt…`.

Reply:
@Logan007 It's my careless. Thank you. :wink:

Reply:
Logan007 is a careful person.😄

Reply:
With all the diverse talents assembled here, we will push it forward! United we are strong! :smile: